### PR TITLE
Add lead partner company name to Contract card

### PR DESCRIPTION
**Pull request by aleksievelizhanin**

### CODE REVIEW

File: app/frontend/src/cljs/teet/contract/contract_common.cljs [hunk 1]
```diff
@@ -57,43 +57,57 @@
...
     [contract-external-links contract]]])
 
+(defn contract-partners-names [contract-with-partners]
+  (let [partners (:company-contract/_contract contract-with-partners)
+        names (reduce
+                (fn [acc item]
+                  (conj acc
+                    (get-in item [:company-contract/company :company/name])))
+                []
+                partners)]
```
Comment (jkorpelin) [inline app/frontend/src/cljs/teet/contract/contract_common.cljs:67]: would `(mapv #(get-in % [:company-contract/company :company/name]))` do the same thing?

File: app/backend/src/clj/teet/contract/contracts_queries.clj [hunk 2]
```diff
@@ -98,28 +98,31 @@
...
+               arglist)
+        contracts-list (->> (d/q {:query {:find '[(pull ?c
+                                     [* {:thk.contract/targets
+                                         [* {:activity/manager
+                                             [:user/given-name :user/family-name]}]}])
+                                                  ?calculated-status]
+                                          :where (into '[[?c :thk.contract/procurement-id _]
+                                                         (contract-status ?c ?calculated-status ?now)] where)
+                                          :in in} :args args})
+                         (mapv contract-db/contract-with-status)
```
Comment (jkorpelin) [inline app/backend/src/clj/teet/contract/contracts_queries.clj:122]: Could fix the indentation.

File: app/frontend/src/cljs/teet/contract/contract_common.cljs [hunk 3]
```diff
@@ -57,43 +57,52 @@
...
+      (when (not-empty (filterv some?
+                         (distinct (mapv #(user-model/user-name (:activity/manager %))
+                                     targets))))
+        [(tr [:contracts :filters :inputs :project-manager])
+         [typography/Paragraph (str/join ", " (filter some?
+                                                (distinct (mapv #(user-model/user-name (:activity/manager %))
+                                                            targets))))]])
+      (when (not-empty partners)
+        [(tr [:fields :company-contract/partners])
+         [typography/Paragraph partners]])
```
Comment (jkorpelin) [inline app/frontend/src/cljs/teet/contract/contract_common.cljs:105]: I don't think we have a field called company-contract/partners? Could use just a normal translation. `[:contracts :company-partners]` or something like that.

File: app/backend/src/clj/teet/contract/contract_db.clj [hunk 4]
```diff
@@ -232,6 +232,27 @@
...
+        lead-partner (get-lead-partner-company db [:thk.contract/procurement-id+procurement-part-id contract-eid])]
+    (assoc contract :thk.contract/lead-partner lead-partner)))
+
+(defn contract-with-all-partners
+  "Used in contract card to show all partners"
+  [db contract]
+  (let [contract-eid (:thk.contract/procurement-id+procurement-part-id contract)
+        partners (get-contract-with-partners db
+                   [:thk.contract/procurement-id+procurement-part-id contract-eid])]
+    (assoc contract :thk.contract/contract-with-partners partners)))
```
Comment (jkorpelin) [inline app/backend/src/clj/teet/contract/contract_db.clj:254]: This fn is passed the whole contract information and in the `get-contract-with-partners` you are pulling all of the contract information again? And associng it to the` contract-with-partners?` key. Should you only fetch the partners or just add the company keys to the pull in the contract listing query. `'[* {:company-contract/_contract [{:company-contract/company [*]}] }]`

**Review by jkorpelin — CHANGES_REQUESTED** [review]

File: app/common/resources/public/language/en.edn [hunk 5]
```diff
@@ -308,6 +308,7 @@
    :filtered-results "Filtered {count} results",
    :results "{count} results",
    :view "View"},
+  :company-partners "Parnters",
```
Comment (erno) [inline app/common/resources/public/language/en.edn:311]:
```suggestion
  :company-partners "Partners",
```

Fixed in the spreadsheet too

File: app/backend/src/clj/teet/contract/contract_db.clj [hunk 6]
```diff
@@ -232,6 +232,13 @@
             db contract-id)
        first))
 
+(defn get-lead-partner-company [db contract-id]
```
Comment (jkorpelin) [inline app/backend/src/clj/teet/contract/contract_db.clj:235]: is this used anywhere?

**Review by jkorpelin — APPROVED** [review]

**Pull request merged**